gdx/src/com/badlogic/gdx/scenes/scene2d/ui/Skin.java

```diff
@@ -142,6 +144,17 @@ public void remove (String name, Class type) {
 		if (type == Sprite.class) return (T)getSprite(name);
 
 		ObjectMap<String, Object> typeResources = resources.get(type);
+		if (typeResources == null) {
+			// Search within derived classes
+			Iterator<Class> typeIterator = resources.keys();
+			while (typeIterator.hasNext()) {
+				Class resourceType = typeIterator.next();
+				if (type.isAssignableFrom(resourceType)) {
+					typeResources = resources.get(resourceType);
+					break;
+				}
+			}
+		}

```

Comment:
With this change you don't have to register the new style from code, you can handle everything in the json. Consider the following:

    com.badlogic.gdx.tests.scenes.scene2d.ui.DynamicChildActorsTest$FlashingLabel$FlashingLabelStyle {
        flashing-label: {period: 2}
    },
    com.badlogic.gdx.scenes.scene2d.ui.TextButton.TextButtonStyle {
        flashing-button: {labelStyle: flashing-label}
    }

You can get a flashing button with a single line of code:

    TextButton textButton2 = new TextButton("TextButton 2", skin, "flashing-button");

This is further explained in the last comment below.